docs/integrations/gitlab/org.md

```diff
@@ -269,67 +269,114 @@ You can inject your own transformation logic to help map GitLab API responses
-
-/**
- * Registers the GitlabDiscoveryEntityProvider with the catalog processing extension point.
- *
- * @alpha
- */
-export const catalogModuleGitlabOrgDiscoveryEntityProvider =
-  createBackendModule({
-    pluginId: 'catalog',
-    moduleId: 'gitlabOrgDiscoveryEntityProvider',
-    register(env) {
-      env.registerInit({
-        deps: {
-          config: coreServices.rootConfig,
-          catalog: catalogProcessingExtensionPoint,
-          logger: coreServices.logger,
-          scheduler: coreServices.scheduler,
-          events: eventsServiceRef,
+To enable this create your own implementation of this plugin, passing a function into the `GitlabOrgDiscoveryEntityProvider`.
+You can pass a `UserTransformer`, a `GroupEntitiesTransformer` or a `GroupNameTransformer` (or all of them).
+The function is invoked for each item (user or group) that is returned from the API.
+
+The example below uses a custom `UserTransformer` based on the default transformer used in [catalog-backend-module-gitlab](https://github.com/backstage/backstage/blob/master/plugins/catalog-backend-module-gitlab/src/lib/defaultTransformers.ts#L99) with the addition of ensuring
+that users without an email address in GitLab are assigned a default email based on their name:
+
+- Create a new module that extends an existing plugin using the backstage cli:
+
+  ```shell
+  yarn backstage-cli new
+  ```
+
+  - Select `backend-module - A new backend module that extends an existing backend plugin with additional features`.
+  - Choose names for the plugin and module. Example: `catalog` and `gitlab-org-discovery-entity-provider`.
+
+- Add the following code to the new module (`module.ts`):
+
+  ```ts
+  import {
+    coreServices,
+    createBackendModule,
+  } from '@backstage/backend-plugin-api';
+  import { catalogProcessingExtensionPoint } from '@backstage/plugin-catalog-node/alpha';
+  import { eventsServiceRef } from '@backstage/plugin-events-node';
+  import {
+    GitlabOrgDiscoveryEntityProvider,
+    UserTransformerOptions,
+  } from '@backstage/plugin-catalog-backend-module-gitlab';
+  import { UserEntityV1alpha1 } from '@backstage/catalog-model';
+
+  function customTransformer(
+    options: UserTransformerOptions,
+  ): UserEntityV1alpha1 {
+    const annotations: { [key: string]: string } = {};
+    annotations[`${options.integrationConfig.host}/user-login`] =
+      options.user.web_url;
+    if (options.user?.group_saml_identity?.extern_uid) {
+      annotations[`${options.integrationConfig.host}/saml-external-uid`] =
+        options.user.group_saml_identity.extern_uid;
+    }
+    const entity = {
+      apiVersion: 'backstage.io/v1alpha1' as const,
+      kind: 'User' as const,
+      metadata: {
+        name: options.user.username,
+        annotations,
+      },
+      spec: {
+        profile: {
+          displayName: options.user.name || void 0,
+          picture: options.user.avatar_url || void 0,
+          email: options.user.email || void 0,
```

Comment:
Is it not better to just do `displayName: options.user.name || undefined` for these?

WDYT?